projects/storefrontlib/cms-components/content/paragraph/anchor.pipe.ts

```diff
@@ -0,0 +1,29 @@
+import { Pipe, PipeTransform, Renderer2 } from '@angular/core';
+import { WindowRef } from '@spartacus/core';
+
+/* Enriches anchor links with the current location (path and query params).
+ * Useful for cms-provided content injected via innerHTML.
+ */
+@Pipe({ name: 'cxAnchor' })
+export class AnchorPipe implements PipeTransform {
+  constructor(protected renderer: Renderer2, protected winRef: WindowRef) {}
+
+  protected getPath(anchorId: string): string {
+    const { pathname, search } = this.winRef.location;
```

Comment:
In SSR, the winRef.location has only those 2 properties:
https://github.com/SAP/spartacus/blob/c347ac28bed48a1a3540bf3e043650ef936bee4b/projects/core/src/window/window-ref.ts#L78-L81
What about:
```
const currentUrlWithoutFragment = this.winRef.document.location.href.replace(/#.*$/, '');
return `${currentUrlWithoutFragment}#${anchorId}`
```